tests/unit/test_tutoring_model.py

```diff
@@ -129,6 +129,241 @@ def test_prepare_les_for_ga(learning_style):
+                "input_value": 0,
+                "processing_dimension": "ref",
+                "processing_value": 9,
+                "understanding_dimension": "seq",
+                "understanding_value": 11,
+            }
+        ),
+        (
+            {
+                "id": 3,
+                "characteristic_id": 4,
+                "perception_dimension": "sns",
+                "perception_value": 1,
+                "input_dimension": "vis",
+                "input_value": 11,
+                "processing_dimension": "act",
+                "processing_value": 1,
+                "understanding_dimension": "glo",
+                "understanding_value": 3,
+            }
+        ),
+    ],
+)
+def test_prepare_les_for_tyche(learning_style):
+    # Test Tyche with success:
+    list_of_les = []
+    list_of_keys = [
+        "ZF",
+        "KÜ",
+        "SE",
+        "LZ",
+        "ZL",
+        "AN",
+        "ÜB",
+        "EK",
+        "FO",
+        "AB",
+        "BE",
+    ]
+    for i, ele_name in enumerate(list_of_keys):
+        le = DM.LearningElement(
+            lms_id=i,
+            activity_type="lesson",
+            classification=ele_name,
+            name="Test LE",
+            university="TH-AB",
+            created_by="Max Mustermann",
+            created_at="2023-09-01",
+        )
+        list_of_les.append(le.serialize())
+    lp = TM.LearningPath(student_id=1, course_id=1, based_on="tyche")
+
+    lp.get_learning_path(
+        student_id=1,
+        learning_style=learning_style,
+        _algorithm="tyche",
+        list_of_les=list_of_les,
+    )
+    result = lp.path
+    erg = False
+    if all(le in result for le in list_of_keys):
+        erg = True
+    assert erg
+
+    tyche_alg = tyche.TycheAlgorithm()
+    last_element = "RQ"
+    tyche_path_success2 = tyche_alg.get_learning_path(
+        learning_style, list_of_les, last_element
+    )
+    print(tyche_path_success2)
+    erg11 = False
+    if all(le in tyche_path_success2 for le in list_of_keys):
+        erg11 = True
+    assert erg11
+
+    list_of_les2 = []
+    list_of_keys2 = [
+        "KÜ",
+        "LZ",
+        "ZL",
+        "ÜB",
+        "FO",
+        "AB",
+        "BE",
+    ]
+    for i, ele_name in enumerate(list_of_keys2):
+        le = DM.LearningElement(
+            lms_id=i,
+            activity_type="lesson",
+            classification=ele_name,
+            name="Test LE",
+            university="TH-AB",
+            created_by="Max Mustermann",
+            created_at="2023-09-01",
+        )
+        list_of_les2.append(le.serialize())
+    lp = TM.LearningPath(student_id=1, course_id=1, based_on="tyche")
+
+    lp.get_learning_path(
+        student_id=1,
+        learning_style=learning_style,
+        _algorithm="tyche",
+        list_of_les=list_of_les2,
+    )
+    result2 = lp.path
+    erg2 = False
+    if all(le in result2 for le in list_of_keys2):
```

Comment:
here maybe assert that result is not None or Unkown

Reply:
additional assert for result2